components/script/dom/abstractworkerglobalscope.rs

```diff
@@ -55,12 +62,98 @@ impl<T: JSTraceable + DomObject + 'static> ScriptChan for WorkerThreadWorkerChan
     }
 }
 
-impl<T: DomObject> ScriptPort for Receiver<(Trusted<T>, WorkerScriptMsg)> {
+impl ScriptPort for Receiver<DedicatedWorkerScriptMsg> {
     fn recv(&self) -> Result<CommonScriptMsg, ()> {
-        match self.recv().map(|(_, msg)| msg) {
-            Ok(WorkerScriptMsg::Common(script_msg)) => Ok(script_msg),
-            Ok(WorkerScriptMsg::DOMMessage(_)) => panic!("unexpected worker event message!"),
-            Err(_) => Err(()),
+        let common_msg = match self.recv() {
+            Ok(DedicatedWorkerScriptMsg::CommonWorker(_worker, common_msg)) => common_msg,
+            Err(_) => return Err(()),
+            _ => panic!("unexpected worker event message!")
+        };
+        match common_msg {
+            WorkerScriptMsg::Common(script_msg) => Ok(script_msg),
+            WorkerScriptMsg::DOMMessage(_) => panic!("unexpected worker event message!"),
         }
     }
 }
+
+pub trait WorkerEventLoopMethods {
+    type TimerMsg: Send;
+    type WorkerMsg: QueuedTaskConversion + Send;
+    type Event;
+    fn timer_event_port(&self) -> &Receiver<Self::TimerMsg>;
+    fn task_queue(&self) -> &TaskQueue<Self::WorkerMsg>;
+    fn handle_event(&self, event: Self::Event);
+    fn handle_worker_post_event(&self, worker: &TrustedWorkerAddress);
+    fn worker_global_scope(&self) -> &WorkerGlobalScope;
+    fn global_scope(&self) -> &GlobalScope;
```

Comment:
thinking of replace this method with one called `perform_microtask_checkpoint`, since that's all that is being done with the global scope as part of running the event-loop...